scripts/mosquitto.sh

```diff
@@ -1,83 +1,58 @@
 #!/bin/bash
 
-set -ex
+# Run mosquitto MQTT broker with configuration needed for the test suite.
+#
+# If mosquitto is installed locally, it will be run. Otherwise, fall back to
+# running a containerized version of mosquitto. The --container/-C option may be
+# supplied to force running a containerized mosquitto.
+#
+# N.B. On MacOS, a native mosquitto executable must be run to connect to
+# mosquitto using a unix domain socket.
 
-CONTAINER_ID="$(docker container ls | grep eclipse-mosquitto | awk {'print $1'})"
-COMMAND="$1"
-SCRIPTS=$(dirname $0)
-ROOT=$(dirname $SCRIPTS)
+set -eu -o pipefail
 
 usage()
 {
-    echo "Usage: mosquitto.sh [start] [stop] [status]"
+    echo "Usage: mosquitto.sh [--container|-C]"
     exit 2
 }
 
-run()
+run-installed-mosquitto()
 {
-    if [ -z "$CONTAINER_ID" ]; then
-        docker run \
-            -p 1883:1883 \
-            -p 1884:1884 \
-            -p 8883:8883 \
-            -p 8080:8080 \
-            -p 8081:8081 \
-            -v $(pwd)/$ROOT/mosquitto/config:/mosquitto/config \
-            -v $(pwd)/$ROOT/mosquitto/certs:/mosquitto/certs \
-            eclipse-mosquitto
-    else
-        echo "Mosquitto is already running"
-    fi
+    mosquitto -c mosquitto/config/mosquitto.conf
 }
 
-start()
+run-containerized-mosquitto()
 {
-    if [ -z "$CONTAINER_ID" ]; then
-        docker run \
-            -d \
-            -p 1883:1883 \
-            -p 1884:1884 \
-            -p 8883:8883 \
-            -p 8080:8080 \
-            -p 8081:8081 \
-            -v $(pwd)/$HOME/../mosquitto/config:/mosquitto/config \
-            -v $(pwd)/$HOME/../mosquitto/certs:/mosquitto/certs \
-            eclipse-mosquitto
-    else
-        echo "Mosquitto is already running"
-    fi
+    docker run \
+        -p 1883:1883 \
+        -p 1884:1884 \
+        -p 8883:8883 \
+        -p 8080:8080 \
+        -p 8081:8081 \
+        -v ./mosquitto/config:/mosquitto/config \
+        -v ./mosquitto/certs:/mosquitto/certs \
```

Comment:
These folder mappings need a full path. It fails otherwise

Reply:
Interesting. The relative paths worked for me, but I'm using podman, not docker.

I changed this to absolute paths in the latest version of the change.